submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/integration/PersistenceContextImpl.kt

```diff
@@ -43,29 +43,30 @@ open class PersistenceContextImpl(
 
     @Transactional(isolation = Isolation.READ_UNCOMMITTED)
     override fun saveSubmission(saveRequest: SaveRequest): ExtSubmission {
-        val (submission, submitter, mode) = saveRequest
-        return lockExecutor.executeLocking(submission.accNo) { saveSubmission(submission, submitter, mode) }
+        val (submission, mode) = saveRequest
+        return lockExecutor.executeLocking(submission.accNo) { saveSubmission(submission, mode) }
     }
 
     @Transactional
     override fun refreshSubmission(submission: ExtSubmission, submitter: User) {
-        saveSubmission(SaveRequest(submission.copy(version = submission.version + 1), submitter, FileMode.MOVE))
+        saveSubmission(SaveRequest(submission.copy(version = submission.version + 1), FileMode.MOVE))
     }
 
-    private fun saveSubmission(submission: ExtSubmission, submitter: User, mode: FileMode): ExtSubmission {
+    private fun saveSubmission(submission: ExtSubmission, mode: FileMode): ExtSubmission {
         subRepository.expireActiveVersions(submission.accNo)
-        deleteSubmissionDrafts(submitter.id, submission.accNo)
-        return saveNewVersion(submission, submitter.email, mode)
+        deleteSubmissionDrafts(submission.submitter, submission.accNo)
```

Comment:
I understand in some cases the owner and the submitter are the same but I'm not sure about deleting the drafts for both. I think we only need to delete it only for the owner which is the one that should have the draft in the first place. If they're the same, then deleting for both won't have any effect.

Reply:
mmm the problem for me is imagining we login in ST as manager or (future reviewer) role, you get a draft created, then original submission owner edit, then manager user check again and because it is a draft it will see not see author changes.